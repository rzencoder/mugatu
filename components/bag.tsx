import { useBag } from '@/context/bagContext'
import { useWishlist } from '@/context/wishlistContext'
import { DeleteIcon, ChevronDownIcon, InfoIcon } from '@chakra-ui/icons'
import {
  Flex,
  Box,
  Link,
  Button,
  useColorMode,
  Menu,
  MenuItem,
  MenuList,
  MenuButton,
} from '@chakra-ui/react'
import Image from 'next/image'
import NextLink from 'next/link'
import { useState, useEffect } from 'react'

const deliveryOptions = [
  {
    name: 'standard delivery',
    price: 4.95,
  },
  {
    name: 'next day delivery',
    price: 6.95,
  },
  {
    name: 'click and collect standard',
    price: 4.95,
  },
  {
    name: 'click and collect next day',
    price: 6.95,
  },
]

const Bag = () => {
  const { bag, updateBag, removeFromBag } = useBag()
  const { addToWishlist } = useWishlist()
  const { colorMode } = useColorMode()
  const [delivery, setDelivery] = useState(deliveryOptions[0])

  const getQuantityOptions = (item) => {
    const options = []
    const size = item.sizes.find((el) => el.size === item.selectedSize)
    for (let i = 1; i <= size.stock; i++) {
      if (i > 10) break
      options.push(i)
    }
    return options
  }

  const calculateSubTotal = () => {
    const total = bag.reduce((acc, cur) => {
      return cur.price * cur.quantity + acc
    }, 0)
    return total.toFixed(2)
  }

  const subtotal = calculateSubTotal()

  const displayDeliveryPrice = (price) => {
    if (price === 0) {
      return 'free'
    } else {
      return `£${price}`
    }
  }

  useEffect(() => {
    if (parseFloat(subtotal) > 50) {
      if (delivery.name === 'standard delivery' || delivery.name === 'click and collect standard') {
        if (delivery.price !== 0) {
          setDelivery({ ...delivery, price: 0 })
        }
      }
    } else {
      if (delivery.price === 0) {
        const originalPrice = deliveryOptions.find((option) => option.name === delivery.name)
        setDelivery({ ...delivery, price: originalPrice.price })
      }
    }
  }, [bag, delivery])

  if (!bag || bag.length === 0) {
    return (
      <Flex direction="column" justify="center" alignItems="center" minHeight="450px">
        <Box
          textTransform="lowercase"
          fontSize="28px"
          fontWeight="600"
          width="80%"
          textAlign="center"
          padding="0 0 30px"
        >
          Your Shopping Bag Is Empty
        </Box>
        <NextLink href="/" passHref>
          <Link>
            <Button>Continue Shopping</Button>
          </Link>
        </NextLink>
      </Flex>
    )
  }

  return (
    <Box bg={colorMode === 'light' ? '#eee' : 'mainBlack'}>
      <Flex
        direction={['column', null, 'row']}
        maxWidth="1000px"
        minHeight="500px"
        margin="0 auto"
        padding={['15px 5px', '20px']}
        justifyContent="space-between"
        position="relative"
      >
        <Flex width={['100%', null, '60%']} direction="column" mr={['5px', '20px']}>
          <Flex
            direction="column"
            p={['15px 10px', '20px']}
            mb="20px"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
          >
            <Box p="0 0 15px" fontWeight="600" fontSize="28px">
              my bag
            </Box>
            {bag.map((item) => {
              return (
                <Flex key={`bag-item-${item.id}`} p="10px 0">
                  <Flex
                    minWidth={['120px', '140px']}
                    maxWidth={['120px', '140px']}
                    maxHeight={['180px', 'none']}
                  >
                    <Image src={item.image.url} width={150} height={225}></Image>
                  </Flex>
                  <Flex
                    direction="column"
                    p={['0 0 0 10px', '0 10px 0 15px']}
                    justifyContent="space-between"
                    width="100%"
                  >
                    <Flex justifyContent="space-between">
                      <NextLink
                        href={`/${item.gender === 'female' ? 'women' : 'men'}/catalog/${item.slug}`}
                        passHref
                      >
                        <Link>
                          <Box fontSize={['16px', '18px']} fontWeight="600">
                            {item.name}
                          </Box>
                        </Link>
                      </NextLink>
                      <Button
                        variant="icon"
                        fontSize="22px"
                        padding="5px"
                        margin={['0 3px', '0 10px']}
                        onClick={() => removeFromBag(item)}
                        color={colorMode === 'light' ? '#999' : '#aaa'}
                      >
                        <DeleteIcon
                          _hover={{
                            color: colorMode === 'light' ? '#777' : '#ccc',
                          }}
                        />
                      </Button>
                    </Flex>
                    <Flex p="5px 0" fontSize={['14px', '15px']} direction="column">
                      <Box p="0">{item.colour}</Box>
                      <Box p="0">size: {item.selectedSize}</Box>
                      <Flex alignItems="center">
                        <Box>qty:</Box>
                        <Menu>
                          <MenuButton
                            fontSize="15px"
                            height="20px"
                            as={Button}
                            rightIcon={<ChevronDownIcon m="0" />}
                            variant="transparentBg"
                            textTransform="lowercase"
                          >
                            {item.quantity}
                          </MenuButton>
                          <MenuList
                            bg={colorMode === 'light' ? 'mainWhite' : 'mainBlack'}
                            fontSize="16px"
                            minWidth="0"
                          >
                            {getQuantityOptions(item).map((el) => {
                              return (
                                <MenuItem
                                  key={`product-quantity-${el}`}
                                  onClick={() => updateBag({ ...item, quantity: el })}
                                >
                                  {el}
                                </MenuItem>
                              )
                            })}
                          </MenuList>
                        </Menu>
                      </Flex>
                    </Flex>
                    <Flex p="5px 0">
                      <Box fontSize={['19px', '22px']} color="mainRed">
                        £{item.price}
                      </Box>
                      <Box
                        fontSize={['14px', '15px']}
                        textDecoration="line-through"
                        margin="3px 0 0 8px"
                        color={colorMode === 'light' ? '#333' : '#ddd'}
                      >
                        £{item.rrp}
                      </Box>
                    </Flex>
                    <Button
                      fontSize={['14px', '15px']}
                      borderWidth="1px"
                      width={['150px', '170px']}
                      height={['35px', '45px']}
                      p={['1px', '3px 5px']}
                      _hover={{ bg: colorMode === 'light' ? 'mainBlackHover' : 'mainWhiteHover' }}
                      onClick={() => {
                        addToWishlist(item)
                        removeFromBag(item)
                      }}
                    >
                      <Box
                        backgroundImage="url(/icons/heart.png)"
                        width={['15px', '20px']}
                        height={['15px', '20px']}
                        backgroundSize="cover"
                        filter={colorMode === 'light' ? 'invert()' : 'none'}
                      />
                      <Box p="0 0 2px 5px" textTransform="lowercase">
                        Move to Wishlist
                      </Box>
                    </Button>
                  </Flex>
                </Flex>
              )
            })}
          </Flex>
          <Flex
            display={['none', null, 'flex']}
            justifyContent="space-between"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
            p="20px"
            fontSize="22px"
            fontWeight="600"
          >
            <Box>sub-total</Box>
            <Box>£{subtotal}</Box>
          </Flex>
        </Flex>
        <Flex
          width={['100%', null, '40%']}
          direction="column"
          height={['350px']}
          bg={colorMode === 'light' ? 'mainWhite' : '#111'}
          p="20px"
          justifyContent="space-between"
        >
          <Box p="0 0 15px" fontWeight="600" fontSize="28px">
            summary
          </Box>
          <Flex direction="column" fontSize="16px">
            <Flex justifyContent="space-between" fontWeight="600">
              <Box>sub-total</Box>
              <Box>£{subtotal}</Box>
            </Flex>
            {subtotal < 50 && (
              <Flex
                bg={colorMode === 'light' ? 'mainBlack' : 'mainWhite'}
                color={colorMode === 'light' ? 'mainWhite' : 'mainBlack'}
                p="8px 12px"
                alignItems="center"
                margin="10px 0"
                fontWeight="600"
              >
                <InfoIcon fontSize="22px" mr="15px" />
                <Box>{`spend £${(50 - parseFloat(subtotal)).toFixed(
                  2
                )} more to qualify for free delivery`}</Box>
              </Flex>
            )}
            <Flex direction="column">
              <Flex justifyContent="space-between" fontWeight="600">
                <Box>delivery</Box>
                <Box>{displayDeliveryPrice(delivery.price)}</Box>
              </Flex>
              <Menu>
                <MenuButton
                  textAlign="left"
                  as={Button}
                  border="none"
                  textTransform="lowercase"
                  p="0"
                  fontWeight="400"
                  fontSize="16px"
                >
                  <Box as="span">{`${delivery.name} (${displayDeliveryPrice(
                    delivery.price
                  )})`}</Box>
                  <ChevronDownIcon />
                </MenuButton>
                <MenuList>
                  {deliveryOptions.map((option, index) => {
                    const price = option.name === delivery.name ? delivery.price : option.price
                    return (
                      <MenuItem
                        key={`delivery-option-${option.name}`}
                        onClick={() => setDelivery(deliveryOptions[index])}
                      >
                        {`${option.name} (${displayDeliveryPrice(price)})`}
                      </MenuItem>
                    )
                  })}
                </MenuList>
              </Menu>
            </Flex>
            <Flex justifyContent="space-between" fontSize="22px" fontWeight="600" p="5px 0">
              <Box>total</Box>
              <Box>£{(parseFloat(subtotal) + delivery.price).toFixed(2)}</Box>
            </Flex>
          </Flex>
          <NextLink href="/checkout" passHref>
            <Link>
              <Button
                bg="#0d6d33"
                border="none"
                color="mainWhite"
                width="100%"
                _hover={{
                  bg: '#15ab51',
                }}
              >
                Checkout
              </Button>
            </Link>
          </NextLink>
        </Flex>
      </Flex>
    </Box>
  )
}

export default Bag
