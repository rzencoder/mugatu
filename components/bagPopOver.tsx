import { useBag } from '@/context/bagContext'
import { DeleteIcon } from '@chakra-ui/icons'
import {
  Box,
  Flex,
  PopoverBody,
  PopoverContent,
  PopoverHeader,
  PopoverCloseButton,
  Button,
  useColorMode,
} from '@chakra-ui/react'
import Image from 'next/image'

const BagPopOver = () => {
  const { colorMode } = useColorMode()
  const { bag, removeFromBag } = useBag()

  const calculateSubTotal = () => {
    const total = bag.reduce((acc, cur) => {
      return cur.price * cur.quantity + acc
    }, 0)
    return total.toFixed(2)
  }

  const calculateNumOfItems = () => {
    return bag.reduce((acc, cur) => {
      return acc + parseInt(cur.quantity)
    }, 0)
  }

  if (!bag || bag.length === 0) {
    return null
  }
  return (
    <PopoverContent
      display={['none', 'block']}
      bg={colorMode === 'light' ? 'mainWhite' : 'mainBlack'}
    >
      <PopoverCloseButton top="12px" />
      <PopoverHeader fontSize="20px" fontWeight="700" textAlign="center">
        my bag
      </PopoverHeader>
      <PopoverBody>
        <Flex direction="column">
          <Box fontWeight="700">{`${calculateNumOfItems()} item${
            calculateNumOfItems() > 1 ? 's' : ''
          }`}</Box>
          <Flex direction="column" height="180px" overflowY="scroll">
            {bag.map((item) => {
              return (
                <Flex key={`popover-bag-item-${item.name}`} p="10px 0">
                  <Flex minWidth="100px">
                    <Image src={item.image.url} width="100" height="150" />
                  </Flex>
                  <Flex
                    direction="column"
                    p="0 10px"
                    fontSize="14px"
                    justifyContent="space-between"
                    textTransform="lowercase"
                  >
                    <Box fontSize="15px">{item.name}</Box>
                    <Flex direction="column" width="80%" fontSize="13px">
                      <Box>{item.colour}</Box>
                      <Box>size: {item.selectedSize}</Box>
                      <Box>qty: {item.quantity}</Box>
                    </Flex>
                    <Flex justifyContent="space-between" alignItems="center">
                      <Box fontWeight="700" fontSize="16px">
                        £{item.price}
                      </Box>
                      <Button
                        variant="transparentBg"
                        onClick={() => {
                          removeFromBag(item)
                        }}
                      >
                        <DeleteIcon />
                      </Button>
                    </Flex>
                  </Flex>
                </Flex>
              )
            })}
          </Flex>
          <Flex justifyContent="space-between" alignItems="center" p="12px 0" fontSize="18px">
            <Box>subtotal</Box>
            <Box fontWeight="700">£{calculateSubTotal()}</Box>
          </Flex>
          <Button>View Bag</Button>
        </Flex>
      </PopoverBody>
    </PopoverContent>
  )
}

export default BagPopOver
