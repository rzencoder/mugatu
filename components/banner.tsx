import { Box, Flex, Link } from '@chakra-ui/react'
import NextLink from 'next/link'
import Image from 'next/image'

interface BannerProps {
  colour: string
  bgColour: string
  imageSrc: string
  title: string
  text: string
  linkHref: string
}

const Banner = ({
  colour,
  bgColour,
  imageSrc,
  title,
  text,
  linkHref,
}: BannerProps): JSX.Element => {
  return (
    <Box
      height={['160px', '200px', '250px', '300px']}
      fontFamily="Montserrat"
      fontSize="80px"
      color="white"
      textAlign="center"
      position="relative"
    >
      <Image className="banner-bg" src={imageSrc} layout="fill" role="presentation" />
      <Box top="0" width="100%" height="100%" bg={bgColour} opacity="0.6" position="absolute"></Box>
      <NextLink href={linkHref} passHref>
        <Link>
          <Flex
            position="absolute"
            top="0"
            height="100%"
            width="100%"
            alignItems="center"
            justifyContent="center"
          >
            <Flex display="column" justifyContent="center" textAlign="center" m="35px 0 15px">
              <Box
                bg={colour}
                transform="skewX(-5deg)"
                fontSize={['24px', '35px', '50px', '60px']}
                p="0 20px"
                width="fit-content"
                m="0 auto"
              >
                {title}
              </Box>
              <Box fontSize={['28px', '40px', '60px', '80px']} m="20px auto 5px">
                {text}
              </Box>
            </Flex>
          </Flex>
        </Link>
      </NextLink>
    </Box>
  )
}

export default Banner
