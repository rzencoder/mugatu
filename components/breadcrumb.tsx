import { Breadcrumb as BreadcrumbContainer, BreadcrumbItem, BreadcrumbLink } from '@chakra-ui/react'
import { useRouter } from 'next/dist/client/router'
import Link from 'next/link'

const Breadcrumb = () => {
  const router = useRouter()

  const getPaths = () => {
    const pathName = router.asPath.substring(1)
    console.log(router)
    const paths = pathName.split('/')
    return ['home', ...paths]
  }

  const paths = getPaths()
  if (paths.length < 3) {
    return null
  }

  const formatLinkHref = (paths, index) => {
    const path = [...paths]
    path.length = index + 1
    if (path.length === 1) return '/'
    path.shift()
    const pathResult = path.map((el) => `/${el}`)
    return pathResult.join('')
  }

  return (
    <BreadcrumbContainer
      margin="15px auto 5px"
      width="90%"
      separator=">"
      display={['none', 'block']}
    >
      {paths.map((path, index) => {
        console.log(path)
        return (
          <BreadcrumbItem key={path} color={index === paths.length - 1 && '#777'}>
            <Link href={formatLinkHref(paths, index)} passHref>
              <BreadcrumbLink>{path.split('-').join(' ')}</BreadcrumbLink>
            </Link>
          </BreadcrumbItem>
        )
      })}
    </BreadcrumbContainer>
  )
}

export default Breadcrumb
