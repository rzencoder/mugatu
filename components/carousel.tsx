import { Box, Flex, Link, useColorMode } from '@chakra-ui/react'
import Image from 'next/image'
import { useProducts } from '@/context/productsContext'
import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'
import Slider from 'react-slick'
import NextLink from 'next/link'

const settings = {
  dots: false,
  infinite: true,
  speed: 500,
  slidesToShow: 4,
  slidesToScroll: 4,
  initialSlide: 0,
  responsive: [
    {
      breakpoint: 992,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 3,
      },
    },
    {
      breakpoint: 767,
      settings: {
        slidesToShow: 2,
        slidesToScroll: 2,
      },
    },
    {
      breakpoint: 360,
      settings: {
        slidesToShow: 1,
        slidesToScroll: 1,
      },
    },
  ],
}

const CarouselComponent = ({ products }) => {
  const { colorMode } = useColorMode()

  return (
    <Box width="80%" margin="30px auto">
      <Box fontSize="30px" textAlign="center" p="20px 5px 30px" fontWeight="700">
        <span role="img" aria-label="">
          ðŸ”¥
        </span>{' '}
        hot right now
      </Box>
      {products.length > 0 && (
        <Slider {...settings}>
          {products.map((product) => {
            return (
              <NextLink
                key={`carousel-${product.name}`}
                href={`/${product.gender === 'female' ? 'women' : 'men'}/catalog/${product.slug}`}
                passHref
              >
                <Link>
                  <Box
                    padding={['0 10px', '0 20px']}
                    textTransform="lowercase"
                    m={['0 auto', '0']}
                    maxWidth="250px"
                  >
                    <Image src={product.image.url} width="200px" height="300px" />
                    <Flex direction="column" maxWidth="200px" p="5px 0">
                      <Box>{product.name}</Box>
                      <Flex justifyContent="flex-start" p="5px 0">
                        <Box color="mainRed" fontSize="19px" fontWeight="700" mr="15px">
                          Â£{product.price}
                        </Box>
                        <Box
                          fontSize="15px"
                          textDecoration="line-through"
                          color={colorMode === 'light' ? '#777' : '#ccc'}
                        >
                          Â£{product.rrp}
                        </Box>
                      </Flex>
                    </Flex>
                  </Box>
                </Link>
              </NextLink>
            )
          })}
        </Slider>
      )}
    </Box>
  )
}

export default CarouselComponent
