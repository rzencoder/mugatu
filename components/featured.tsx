import { Box, Flex, Link, useColorMode } from '@chakra-ui/react'
import Image from 'next/image'
import NextLink from 'next/link'

export default function Featured({ items, gender }): JSX.Element {
  const { colorMode } = useColorMode()

  return (
    <Box width="100%" position="relative">
      <Box
        fontFamily="Permanent Marker"
        fontSize={['34px', '40px', '50px', '60px']}
        p="20px 0 0"
        textAlign="center"
        position="relative"
        zIndex="10"
        color={gender === 'female' ? '#ec2392' : '#222'}
      >
        <Box
          height="fit-content"
          bg="mainWhite"
          width="fit-content"
          p="0px 30px"
          margin="0 auto"
          transform="rotate(
-2deg
)"
        >
          shop by style
        </Box>
      </Box>
      <Box position="absolute" top="0" height="100%" width="100%" opacity="0.15">
        <Image src="/landing/women/women-bg.jpg" layout="fill" className="featured-bg" />
      </Box>
      <Flex wrap="wrap" justifyContent="center" width="100%" p={['20px 5px', '20px 15px']}>
        {items.map((item, index) => {
          return (
            <Flex
              width={['42%', '40%', null, '21%']}
              direction="column"
              margin={['20px 10px', '20px']}
              key={item.name}
              bg="#eee"
              alignItems="center"
              transform={`rotate(${item.rotate}deg)`}
              boxShadow="6px 6px 6px #111"
              position="relative"
            >
              <NextLink href={item.link} passHref>
                <Link>
                  <Box padding={['10px 10px 0', '20px 20px 0']} filter="saturate(0.7)">
                    <Image width={400} height={500} src={item.imageUrl} alt={item.title} />
                  </Box>
                  <Box
                    color={gender === 'female' ? '#ec2392' : '#222'}
                    fontFamily="Permanent Marker"
                    fontSize={['28px', '34px', '34px', '30px']}
                    textAlign="center"
                    p={['5px', '0 10px 10px']}
                  >
                    {item.name}
                  </Box>
                  {gender === 'female' && (
                    <Box
                      display={index % 2 === 0 ? 'none' : 'block'}
                      position="absolute"
                      bottom={['75px', null, '50px', '70px']}
                      right={`${index === 1 ? '10' : '50'}%`}
                      transform={`rotate(${index === 1 ? '-10' : '60'}deg)`}
                      backgroundImage="url(/landing/women/lipstick.png)"
                      width="40%"
                      backgroundSize="contain"
                      backgroundRepeat="no-repeat"
                      opacity="0.6"
                      height={['20%']}
                    ></Box>
                  )}
                </Link>
              </NextLink>
            </Flex>
          )
        })}
      </Flex>
    </Box>
  )
}
