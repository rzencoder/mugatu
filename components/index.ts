export { default as Navigation } from './navigation'
export { default as Footer } from './footer'
export { default as FullProduct } from './fullProduct'
export { default as Jumbotron } from './jumbotron'
export { default as Featured } from './featured'
export { default as Filter } from './filter'
export { default as Sort } from './sort'
export { default as ImageInfo } from './imageInfo'
export { default as Products } from './products'
export { default as Loader } from './loader'
