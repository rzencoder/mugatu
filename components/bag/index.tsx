import { useBag } from '@/context/bagContext'
import { calculateSubTotal } from '@/utils/bag'
import { Flex, Box, useColorMode } from '@chakra-ui/react'
import BagItem from './bagItem'
import EmptyBag from './emptyBag'
import Summary from './summary'

const Bag = () => {
  const { bag } = useBag()
  const { colorMode } = useColorMode()
  const subtotal = calculateSubTotal(bag)

  if (!bag || bag.length === 0) {
    return <EmptyBag />
  }

  return (
    <Box bg={colorMode === 'light' ? '#eee' : 'mainBlack'}>
      <Flex
        direction={['column', null, 'row']}
        maxWidth="1000px"
        minHeight="500px"
        margin="0 auto"
        padding={['15px 5px', '20px']}
        justifyContent="space-between"
        position="relative"
      >
        <Flex width={['100%', null, '60%']} direction="column" mr={['5px', '20px']}>
          <Flex
            direction="column"
            p={['15px 10px', '20px']}
            mb="20px"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
          >
            <Box p="0 0 15px" fontWeight="600" fontSize="28px">
              my bag
            </Box>
            {bag.map((item) => {
              return <BagItem key={`bag-item-${item.id}`} item={item} />
            })}
          </Flex>
          <Flex
            display={['none', null, 'flex']}
            justifyContent="space-between"
            bg={colorMode === 'light' ? 'mainWhite' : '#111'}
            p="20px"
            fontSize="22px"
            fontWeight="600"
          >
            <Box>sub-total</Box>
            <Box>Â£{subtotal}</Box>
          </Flex>
        </Flex>
        <Summary subtotal={subtotal} />
      </Flex>
    </Box>
  )
}

export default Bag
