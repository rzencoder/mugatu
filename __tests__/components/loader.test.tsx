import React from 'react'
import { render } from '@testing-library/react'
import { Loader } from '../../components/'

describe('<Loader />', () => {
  it('renders <Loader /> component', () => {
    const { container } = render(<Loader />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
