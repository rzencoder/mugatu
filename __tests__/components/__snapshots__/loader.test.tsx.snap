// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> renders <Loader /> component 1`] = `
<div
  class="css-17r1j6n"
>
  <div
    class="chakra-spinner css-1y7joxr"
  >
    <span
      class="css-f8n5zr"
    >
      Loading...
    </span>
  </div>
</div>
`;
